### ISSUE TITLE
any plans for updating to smile 3.0.x ?

### ISSUE THREAD

**Issue by behrica** [generateme/fastmath#26]
I was looking into it for `scicloj.ml.smile`.
But as it depends on fastmath, I think it would be better to upgrade fastmath before.

**Comment by genmeblog** [comment]
Yes, there are such plans. I'll try to figure out how to make it properly with minimal effort. The main point of changes is around clustering API (no more double-double arrays input for some functions).

**Comment by behrica** [comment]
yes, I saw other big changes, but not affecting fastmath I think:
https://github.com/scicloj/scicloj.ml.smile/issues/12

But I use indeed the clustering from fastmath:
https://github.com/scicloj/scicloj.ml.smile/blob/main/src/scicloj/ml/smile/clustering.clj

**Comment by genmeblog** [comment]
Indeed... I was fooled by changes in classification. I'll take a look soon and upgrade it.

**Comment by genmeblog** [comment]
Looks like it went smoothly. Some minor API changes in my case:
* `setSeed` can't be called without arguments
* `nrows` -> `nrow` in Matrix class
Roughly tested and all other things worked (clustering), so expect a snapshot soon.

**Comment by genmeblog** [comment]
BTW, all the SMILE 3.0.2 deps are:

```clojure
[com.github.haifengl/smile-base "3.0.2"]
                 [com.github.haifengl/smile-core "3.0.2"]
                 [com.github.haifengl/smile-mkl "3.0.2"]
                 
                 [org.bytedeco/arpack-ng "3.9.0-1.5.9"]
                 [org.bytedeco/arpack-ng-platform "3.9.0-1.5.9"]
                 [org.bytedeco/openblas "0.3.23-1.5.9"]
                 [org.bytedeco/openblas-platform "0.3.23-1.5.9"]
                 [org.bytedeco/javacpp "1.5.9"]
```

**Comment by genmeblog** [comment]
Released 2.3.0 with Smile 3.0.2
